src/solvers/sat/satcheck_cadical.cpp

```diff
@@ -120,13 +135,21 @@ satcheck_cadicalt::~satcheck_cadicalt()
 
 void satcheck_cadicalt::set_assumptions(const bvt &bv)
 {
-  INVARIANT(false, "method not supported");
+  assumptions = bv;
+
+  for(const auto &a : assumptions)
+  {
+    if(a.is_true())
+    {
+      assumptions.clear();
```

Comment:
:no_entry_sign: I don't understand this. Assumptions are connected conjunctively.

Reply:
That was me copy&pasting from the `satcheck_minisat2.cpp` ... probably worth reviewing.

Reply:
See https://github.com/diffblue/cbmc/pull/4810